### ISSUE TITLE
Transitioning.View breaks on android when children has images inside it

### ISSUE THREAD

**Issue by victorrseloy** [software-mansion/react-native-reanimated#657]
I have a layout with three images and basically want to change the parent orientation from row to column, on IOS it works perfectly but on Android the transition api is not behaving as expected.

IOS:
![2020-03-16 19 42 42](https://user-images.githubusercontent.com/819286/76790542-5757ae80-67bf-11ea-8e04-2fad1f6d5aa3.gif)

Android:
![2020-03-16 20 09 13](https://user-images.githubusercontent.com/819286/76791876-13b27400-67c2-11ea-90ef-fbd0a1e95df6.gif)

Here follows the code, the issue is reproducible on this snack too https://snack.expo.io/@victorrseloy/frowning-salsa

```
import React, { FunctionComponent, useState, useRef } from "react";
import {
  Transitioning,
  Transition,
  TransitioningView
} from "react-native-reanimated";
import { StyleSheet, Text, View, Image, TouchableOpacity } from "react-native";

const layouts = [
  {
    id: 0,
    layout: { flex: 1, flexDirection: "column" },
    child: {
      backgroundColor: "white",
      marginVertical: 10,
      marginHorizontal: 20,
      borderRadius: 5,
      flex: 1
    }
  },
  {
    id: 1,
    layout: { flex: 1, flexDirection: "row" },
    child: {
      backgroundColor: "white",
      marginVertical: 20,
      marginHorizontal: 5,
      height: 70,
      borderRadius: 5,
      flex: 1
    }
  }
];

const images = [
  "https://picsum.photos/id/142/300/500",
  "https://picsum.photos/id/155/300/500",
  "https://picsum.photos/id/239/300/500"
];

const animationCallbackFactory = (layoutId, setSelectedLayout, ref) => () => {
  setSelectedLayout(layoutId);
  if (ref) {
    ref.current.animateNextTransition();
  }
};

const transition = (
  <Transition.Change durationMs={400} interpolation="easeInOut" />
);

const ThreeLayouts: FunctionComponent = () => {
  const [selectedLayout, setSelectedLayout] = useState(0);
  const ref = useRef<TransitioningView>();
  return (
    <View style={styles.container}>
      <Transitioning.View
        ref={ref}
        transition={transition}
        style={layouts[selectedLayout].layout}
      >
        {images.map(item => (
          <View key={item} style={[layouts[selectedLayout].child]}>
            <Image style={styles.cardImage} source={{ uri: item }} />
          </View>
        ))}
      </Transitioning.View>
      <View>
        {layouts.map(currentLayout => (
          <TouchableOpacity
            style={[
              styles.select,
              selectedLayout == currentLayout.id ? styles.selected : {}
            ]}
            onPress={animationCallbackFactory(
              currentLayout.id,
              setSelectedLayout,
              ref
            )}
          >
            <Text style={styles.text}>Layout One</Text>
          </TouchableOpacity>
        ))}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  cardImage: {
    resizeMode: "cover",
    flex: 1,
    width: null,
    height: null
  },
  select: {
    flexDirection: "row",
    textAlign: "center",
    backgroundColor: "#52BCF9",
    justifyContent: "space-around",
    alignItems: "center",
    height: 40,
    borderBottomColor: "white",
    borderBottomWidth: 1
  },
  selected: {
    backgroundColor: "#C3423F"
  },
  text: { color: "white", fontSize: 20 }
});

export default ThreeLayouts;

```

**Comment by MartinGrue** [comment]
This works for me on android:
```ts
import React, { FunctionComponent, useState, useRef } from "react";
import Animated, {
  Transitioning,
  Transition,
  TransitioningView
} from "react-native-reanimated";
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  ViewStyle,
  ImageStyle
} from "react-native";

interface Layout {
  id: string;
  layout: {
    container: ViewStyle;
    child?: ImageStyle;
  };
}
const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  cardImage: {
    resizeMode: "cover",
    flex: 1,
  },
  select: {
    flexDirection: "row",
    textAlign: "center",
    backgroundColor: "#52BCF9",
    justifyContent: "space-around",
    alignItems: "center",
    height: 40,
    borderBottomColor: "white",
    borderBottomWidth: 1
  },
  selected: {
    backgroundColor: "#C3423F"
  },
  text: { color: "white", fontSize: 20 }
});
const layouts: Layout[] = [
  {
    id: "0",

    layout: {
      container: {
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center"
      },
      child: {
        backgroundColor: "white",
        marginVertical: 10,
        marginHorizontal: 20,
        borderRadius: 5,
        width: "100%"
      }
    }
  },
  {
    id: "1",

    layout: {
      container: { flexDirection: "row" },
      child: {
        backgroundColor: "white",
        marginVertical: 20,
        marginHorizontal: 5,
        height: 70,
        borderRadius: 5
      }
    }
  }
];

const images = [
  {
    id: 0,
    source: require("./142-300x500.jpg")
  },
  {
    id: 1,
    source: require("./155-300x500.jpg")
  },
  {
    id: 2,
    source: require("./239-300x500.jpg")
  }
];
const transition = (
  <Transition.Change durationMs={400} interpolation="easeInOut" />
);

const ThreeLayouts: FunctionComponent = () => {
  const [selectedLayout, setLayout] = useState(layouts[0].layout);
  const ref = useRef<TransitioningView>(null);
  return (
    <View style={styles.container}>
      <Transitioning.View
        ref={ref}
        transition={transition}
        style={[styles.container, selectedLayout.container]}
      >
        {images.map(item => (
          <Animated.Image
            key={item.id}
            source={item.source}
            style={[selectedLayout.child, styles.cardImage]}
          ></Animated.Image>
        ))}
      </Transitioning.View>
      <View>
        {layouts.map(({ layout, id }) => (
          <TouchableOpacity
            key={id}
            style={[
              styles.select,
              selectedLayout == layout ? styles.selected : {}
            ]}
            onPress={() => {
              if (ref.current) {
                ref.current.animateNextTransition();
              }
              setLayout(layout);
            }}
          >
            <Text style={styles.text}>Layout One</Text>
          </TouchableOpacity>
        ))}
      </View>
    </View>
  );
};

export default ThreeLayouts;
```
Sry i kinda rewrote your code...

I used <Animated.Image> as a substitute <image>

**Comment by victorrseloy** [comment]
Hi @MartinGrue , thanks for your input. The issue was not exactly a question about how to achieve the desired result, but to point out the difference and behaviour between the  platforms and that nothing in the documentation points that the particular application didn't work. 

The API should behave consistently in all platforms, if you try the same thing I posted with LayoutAnimations for example the results are consistent. Another point is that in your example <Animated.Image> isn't really what makes it work as if you replace it by plain react native <Image> it keeps working. 

This inconsistence probably comes from some bug in the Android native implementation, but I'm not 100% sure though.

**Comment by victorrseloy** [comment]
@jakub-gonet any particular reason why you removed the `can-repro` label? I can offer more details and post more data here to help to reproduce if required if this snack is not enough https://snack.expo.io/@victorrseloy/frowning-salsa

**Comment by jakub-gonet** [comment]
@victorrseloy, no, just miss clicked, it should keep its label, I'll correct it.

**Comment by jakub-gonet** [comment]
We discussed Transitions internally in our team. Considering that v2 makes creating animations easier and Transitions was an experimental API for a long time, we decided to leave Transitions as they are.
 
We won’t actively fix bugs in them, but we’ll happily merge any PR with fixes. We’re also thinking about rewriting Transitions in v2 sometime in the future.

I’m closing this issue, but if you have further questions, I will be happy to help.
